src/app/shared/filter/canned-range.component.ts

```diff
@@ -52,65 +53,75 @@ export class CannedRangeComponent implements OnInit, OnDestroy {
+    this.whenOptions = [[TODAY, {range: [1, 'days'], beginDate: utcBeginDate.toDate(), endDate: utcEndDate.toDate()}]];
 
     utcBeginDate.subtract(daysIntoWeek, 'days');
-    this.whenOptions.push([THIS_WEEK, {beginDate: utcBeginDate.toDate(), endDate: utcEndDate.toDate()}]);
+    this.whenOptions.push([THIS_WEEK, {range: [1, 'weeks'], beginDate: utcBeginDate.toDate(), endDate: utcEndDate.toDate()}]);
 
     utcBeginDate.subtract(7, 'days');
-    this.whenOptions.push([TWO_WEEKS, {beginDate: utcBeginDate.toDate(), endDate: utcEndDate.toDate()}]);
+    if (this.filter.interval !== INTERVAL_15MIN ||
+      (this.filter.interval === INTERVAL_15MIN && !isMoreThanXDays(10, utcBeginDate, utcEndDate))) {
+      this.whenOptions.push([TWO_WEEKS, {range: [2, 'weeks'], beginDate: utcBeginDate.toDate(), endDate: utcEndDate.toDate()}]);
+    }
 
-    utcBeginDate = this.beginningOfTodayUTC().date(1);
-    this.whenOptions.push([THIS_MONTH, {beginDate: utcBeginDate.toDate(), endDate: utcEndDate.toDate()}]);
+    utcBeginDate = beginningOfTodayUTC().date(1);
+    if (this.filter.interval !== INTERVAL_15MIN ||
+      (this.filter.interval === INTERVAL_15MIN && !isMoreThanXDays(10, utcBeginDate, utcEndDate))) {
+      this.whenOptions.push([THIS_MONTH, {range: [1, 'months'], beginDate: utcBeginDate.toDate(), endDate: utcEndDate.toDate()}]);
+    }
 
     utcBeginDate.subtract(2, 'months');
-    this.whenOptions.push([THREE_MONTHS, {beginDate: utcBeginDate.toDate(), endDate: utcEndDate.toDate()}]);
+    if (this.filter.interval === INTERVAL_DAILY ||
+      (this.filter.interval === INTERVAL_15MIN && !isMoreThanXDays(10, utcBeginDate, utcEndDate)) ||
+      (this.filter.interval === INTERVAL_HOURLY && !isMoreThanXDays(40, utcBeginDate, utcEndDate))) {
+      this.whenOptions.push([THREE_MONTHS, {range: [3, 'months'], beginDate: utcBeginDate.toDate(), endDate: utcEndDate.toDate()}]);
+    }
 
-    utcBeginDate = this.beginningOfTodayUTC().month(0).date(1);
-    this.whenOptions.push([THIS_YEAR, {beginDate: utcBeginDate.toDate(), endDate: utcEndDate.toDate()}]);
+    utcBeginDate = beginningOfTodayUTC().month(0).date(1);
+    if (this.filter.interval === INTERVAL_DAILY ||
+      (this.filter.interval === INTERVAL_15MIN && !isMoreThanXDays(10, utcBeginDate, utcEndDate)) ||
+      (this.filter.interval === INTERVAL_HOURLY && !isMoreThanXDays(40, utcBeginDate, utcEndDate))) {
+      this.whenOptions.push([THIS_YEAR, {range: [1, 'year'], beginDate: utcBeginDate.toDate(), endDate: utcEndDate.toDate()}]);
+    }
 
     utcEndDate.subtract(1, 'days');
-    utcBeginDate = this.beginningOfTodayUTC().subtract(1, 'days');
-    this.whenOptions.push([YESTERDAY, {beginDate: utcBeginDate.toDate(), endDate: utcEndDate.toDate()}]);
+    utcBeginDate = beginningOfTodayUTC().subtract(1, 'days');
+    this.whenOptions.push([YESTERDAY, {range: [1, 'days'], beginDate: utcBeginDate.toDate(), endDate: utcEndDate.toDate()}]);
```

Comment:
This new `range` business fixes the problem with overriding date ranges where the dates match but the named ranges don't, i.e. 'This week' is the same as 'This month' because the 1st was a Sunday.